contributor-docs/testing.md

```diff
@@ -5,19 +5,88 @@
 <!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
 ## Table of Contents
 
-- [Overview](#overview)
-- [Prerequisites](#prerequisites)
-- [Components](#components)
-  - [Visual Regression Testing](#visual-regression-testing)
-  - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Strategy](#strategy)
+- [Unit Tests](#unit-tests)
+  - [Overview](#overview)
+  - [What We Test](#what-we-test)
+  - [Running Tests](#running-tests)
+- [Visual Regression Tests](#visual-regression-tests)
+  - [Overview](#overview)
+  - [Prerequisites](#prerequisites)
+  - [Components](#components)
+    - [Visual Regression Testing](#visual-regression-testing)
+    - [Accessibility Verification Testing](#accessibility-verification-testing)
+- [Interaction Tests](#interaction-tests)
+  - [As A Part Of Unit Tests](#as-a-part-of-unit-tests)
+  - [Storybook Interaction Tests](#storybook-interaction-tests)
+- [Accessibility Tests](#accessibility-tests)
 - [Continous Integration](#continous-integration)
 - [FAQ](#faq)
   - [Why am I seeing `browserType.launch: Executable doesn't exist at ../path`?](#why-am-i-seeing-browsertypelaunch-executable-doesnt-exist-at-path)
 
 <!-- END doctoc generated TOC please keep comment here to allow auto update -->
 <!-- prettier-ignore-end -->
 
-## Overview
+##Strategy
+
+Strategy
+
+##Unit Tests
+
+###Overview
+
+We use [Jest](https://jestjs.io/) as our test runner to write and run our unit tests. Our prefered way of structuring tests is with `describe/it` block format and they are authored within the components's source directory with the file name of `componentName.test.tsx`
+
+We predominantly use [@testing-library/react](https://testing-library.com/docs/react-testing-library/intro/)'s testing helpers such as
```

Comment:
It might be worth mentioning that we are trying to move away from [enzyme](https://enzymejs.github.io/enzyme/). What do you think?

Reply:
Great point! Yes agreed 🙌🏼  We might as well mention that we are trying to move away from [react-test-renderer](https://reactjs.org/docs/test-renderer.html), at least this is what I gathered? `render` from `@testing-library/react` is much more enhanced and [has accessible queries](https://testing-library.com/docs/queries/about/#priority).

Reply:
I covered @rezrah 's point about moving away from enzyme more like us moving towards behavioural testing paradigm in the strategy and @joshblack has [a PR](https://github.com/primer/react/pull/2584) to remove the existing enzyme tests 🎉 So I didn't explicitly mentioned about it here. Let me know if you have any concerns or would like to add anything 😊